src/api/java/com/minecolonies/api/util/BlockPosUtil.java

```diff
@@ -229,6 +233,53 @@ public static boolean isPositionSafe(@NotNull final World sender, final BlockPos
                  && sender.getWorldBorder().contains(blockPos);
     }
 
+
+    /**
+     * this find the building that the BlockPos bellong to
+     *
+     * @param world   the world
+     * @param pos for the current block
+     */
+    @Nullable
+    public static IBuilding findBuilding(final World world, final BlockPos pos)
+    {
+        final IColony colony = IColonyManager.getInstance().getColonyByPosFromWorld(world, pos);
+        if(colony == null)
+        {
+            return null;
+        }
+
+        for(IBuilding building : colony.getBuildingManager().getBuildings().values())
```

Comment:
afaik there is a building.isInBuilding or so check you can use already. No need for all of this